poc/four/load_student_data.py

```diff
@@ -0,0 +1,370 @@
+            response.raise_for_status()
+            token_response = response.json()
+            return token_response.get('access_token')
+        except Exception as e:
+            print(f"✗ Error getting OAuth token: {e}")
+            return None
+
+
+class StudentDataLoader:
+    """Loads student data from CSV and posts to Ed-Fi API"""
+    
+    def __init__(self, discovery_client: EdFiDiscoveryClient):
+        self.discovery_client = discovery_client
+        self.access_token = None
+    
+    def authenticate(self, client_id: str, client_secret: str) -> bool:
+        """Authenticate with the Ed-Fi API"""
+        print("\nAuthenticating with Ed-Fi API...")
+        self.access_token = self.discovery_client.get_oauth_token(client_id, client_secret)
+        
+        if self.access_token:
+            print("✓ Successfully authenticated")
+            return True
+        else:
+            print("✗ Authentication failed")
+            return False
+    
+    def load_student_data(self, csv_file_path: str) -> List[Dict[str, Any]]:
+        """Load student data from CSV file"""
+        students = []
+        
+        try:
+            with open(csv_file_path, 'r', newline='', encoding='utf-8') as csvfile:
+                reader = csv.DictReader(csvfile)
+                for row in reader:
+                    students.append(row)
+            
+            print(f"✓ Loaded {len(students)} student records from {csv_file_path}")
+            return students
+            
+        except Exception as e:
+            print(f"✗ Error loading CSV file: {e}")
+            return []
+    
+    def transform_student_for_api(self, student_row: Dict[str, str]) -> Dict[str, Any]:
+        """Transform CSV student data to Ed-Fi API format"""
+        student_data = {
+            "studentUniqueId": student_row['uniqueId'],
+            "birthDate": student_row['birthDate'],
+            "firstName": student_row['firstName'],
+            "lastSurname": student_row['lastName']
+        }
+        
+        # Add optional fields if present
+        if student_row.get('middleName'):
+            student_data["middleName"] = student_row['middleName']
+        
+        if student_row.get('preferredFirstName'):
+            student_data["personalTitlePrefix"] = student_row.get('title', '')
+        
+        return student_data
+    
+    def transform_enrollment_for_api(self, student_row: Dict[str, str]) -> Dict[str, Any]:
+        """Transform CSV enrollment data to Ed-Fi API format"""
+        enrollment_data = {
+            "studentReference": {
+                "studentUniqueId": student_row['uniqueId']
+            },
+            "schoolReference": {
+                "schoolId": int(student_row['schoolId'])
+            },
+            "entryDate": student_row['enrollmentDate'],
+            "entryGradeLevelDescriptor": f"uri://ed-fi.org/GradeLevelDescriptor#{student_row['enrollmentGradeLevel'].replace(' ', '%20')}"
```

Comment:
This is the first POC that tries to replace spaces in the descriptor with URL encoded `%20`, which is actually incorrect in the Ed-Fi API. Good tip to offer in a future prompt.